src/transports/tcp.rs

```diff
@@ -242,7 +236,11 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
                                 NEW_CONNECTION => {
                                     loop {
                                         let (stream, address) = match server.accept() {
-                                            Ok((stream, address)) => (stream, address),
+                                            Ok((mut stream, address)) => {
+                                                let _ = poll.registry().deregister(&mut stream);
```

Comment:
not sure about that. What's happening if the remote peer close the stream ?

Reply:
I'm actually not sure why this would be needed. In my understanding, we register the TcpListener with the poll, and we never register individual TcpStreams with it. So we shouldn't need to deregister them? @damip

Reply:
(However, this probably does not introduce bugs, as this call only says "we don't care about the events from this stream anymore". If the stream is closed by the peer, then we no longer receive events from it anyway).

Reply:
> and we never register individual TcpStreams with it
true but it's a mio TcpStream that is registered by mio when created